pandas/core/indexes/datetimes.py

```diff
@@ -56,12 +53,50 @@ def _new_DatetimeIndex(cls, d):
             warnings.simplefilter("ignore")
             result = cls.__new__(cls, verify_integrity=False, **d)
 
-    if tz is not None:
-        result = result.tz_localize('UTC').tz_convert(tz)
     return result
 
 
-class DatetimeIndex(DatetimeArray, DatetimeIndexOpsMixin, Int64Index):
+class DatetimeDelegateMixin(DatetimelikeDelegateMixin):
+    _extra_methods = [
```

Comment:
I started a branch that cleans this up a bit by adding methods to `_datetimelike_ops` and `_datetimelike_methods`. This makes the delegator nice

```python

class DatetimeDelegateMixin(DatetimelikeDelegateMixin):
    _delegated_properties = DatetimeArray._datetimelike_ops
    _delegated_methods = DatetimeArray._datetimelike_methods
    _raw_properties = {
        'date',
        'time',
        'timetz',
        '_box_func',
    } | set(DatetimeArray._bool_ops)
    _raw_methods = {
        'to_pydatetime',
        '_local_timestamps',
        '_has_same_tz'
    }
    _delegate_class = DatetimeArray
```

but causes some less pleasant changes elsewhere. We add things like `_to_perioddelta` to the public API via the `.dt` accessor, some of the tests aren't written to handle things like `_box_func`. Not sure I'll go through with it.